app/connectors/GetCalculationListConnector.scala

```diff
@@ -26,9 +26,10 @@ import scala.concurrent.{ExecutionContext, Future}
 
 class GetCalculationListConnector @Inject()(http: HttpClient, val appConfig: AppConfig)(implicit ec: ExecutionContext) extends DesConnector {
 
-  def calcList(nino: String, taxYear: String, optionalTaxYear: Boolean)(implicit hc: HeaderCarrier): Future[GetCalculationListResponse] = {
-    val getCalcListUrl: String = appConfig.desBaseUrl +
-      s"/income-tax/list-of-calculation-results/$nino" + (if(optionalTaxYear) s"?taxYear=$taxYear" else "")
+  def calcList(nino: String, taxYear: Option[String])(implicit hc: HeaderCarrier): Future[GetCalculationListResponse] = {
+
+    val getCalcListUrl: String =
+      s"${appConfig.ifBaseUrl}/income-tax/list-of-calculation-results/$nino${taxYear.fold("")(year => s"?taxYear=$year")}"
```

Comment:
If this should be IF now rather than DES should use IFConnector instead of DESConnector